### PR TITLE
Fix clippy warnings in tests

### PR DESCRIPTION
**Pull request by eduardosm**
Tested using
```
cargo clippy --all-targets
```

### CODE REVIEW

File: base/src/bit.rs [hunk 1]
```diff
@@ -407,7 +407,9 @@ impl FloatEncoding for f64 {
     }
 }
 
+#[allow(clippy::approx_constant)]
```
Comment (cmpute) [inline base/src/bit.rs:410]: Duplicated attribute. Besides I think we can just substitute related constants such as `core::f64::const::LOG2_10` in the tests.

File: base/src/bit.rs [hunk 2]
```diff
@@ -436,7 +438,7 @@ mod tests {
             f32::from_bits(0x7fffff), // largest subnormal number
             f32::from_bits(0x800000), // smallest normal number
             -123.4567,
-            3.14159265,
+            3.1415927,
```
Comment (cmpute) [inline base/src/bit.rs:441]: Here we can just use `core::f32::const::PI`

File: integer/tests/log.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::approx_constant)]
```
Comment (cmpute) [inline integer/tests/log.rs:1]: Same here, it's better to just substitute the `core::f64::consts::*` to follow the clippy instructions

File: rational/tests/convert.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::approx_constant)]
```
Comment (cmpute) [inline rational/tests/convert.rs:1]: Same, should use `core::f32::consts::PI` in this file. But actually the line triggering this lint is a debug line I accidentally put into the test. Could you please help turning that into an actual test?

**Review by cmpute — COMMENTED** [review]
LGTM other than that some `#![allow(..)]` attributes are unnecessary. Thanks for your PR!

**Comment by eduardosm** [comment]
Done, I removed all `allow(clippy::approx_constant)` and used standard constant definitions instead.

File: rational/tests/convert.rs [hunk 5]
```diff
@@ -54,11 +52,11 @@ fn test_from_f32() {
     assert_eq!(RBig::try_from(2.25f32), Ok(rbig!(9 / 4)));
     assert_eq!(RBig::try_from(-2.25e4f32), Ok(rbig!(-22500)));
     assert_eq!(RBig::try_from(1.1773109e-2f32), Ok(rbig!(12345 / 1048576)));
+    assert!(RBig::try_from(core::f32::consts::PI).is_ok());
 
     assert_eq!(RBig::try_from(f32::INFINITY), Err(OutOfBounds));
     assert_eq!(RBig::try_from(f32::NEG_INFINITY), Err(OutOfBounds));
     assert_eq!(RBig::try_from(f32::NAN), Err(OutOfBounds));
-    dbg!(RBig::try_from(3.1415926f32).unwrap());
```
Comment (eduardosm) [inline rational/tests/convert.rs:59]: I also did this while I was at it, but I'm not sure if you had that in mind for that case

**Comment by cmpute** [comment]
Thanks! I also changed the clippy args in the github workflow. Currently the `dashu-macro` crate is excluded, when your another PR is merged, we should be able to test on that crate as well.

**Pull request merged**